### PR TITLE
(fleet) add minimal e2e coverage for the installer daemon

### PR DESCRIPTION
**Pull request by arbll**
This PR:
- fixes 2 remaining bugs preventing the installer daemon from starting without manual help
- adds an e2e to ensure that behavior

### CODE REVIEW

File: pkg/config/setup/config_nix.go [hunk 1]
```diff
@@ -14,10 +14,13 @@ var (
 	// InstallPath is the default install path for the agent
 	// It might be overridden at build time
 	InstallPath = "/opt/datadog-agent"
+
+	// defaultRunPath is the default path for the agent's runtime files
+	// It might be overridden at build time
+	defaultRunPath = "/opt/datadog-agent/run"
```
Comment (arbll) [inline pkg/config/setup/config_nix.go:20]: Golang does not support overriding fields that are resolved at runtime so the override wasn't working

File: pkg/fleet/installer/service/datadog_installer.go [hunk 2]
```diff
@@ -181,7 +181,16 @@ func getAgentIDs() (uid, gid int, err error) {
...
+	_, err = os.Stat("/etc/datadog-agent/datadog.yaml")
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	// this is expected during a fresh install with the install script / asible / chef / etc...
+	// the config is populated afterwards by the install method and the agent is restarted
+	if !os.IsNotExist(err) {
+		return startUnit(ctx, installerUnit)
+	}
+	return nil
```
Comment (BaptisteFoy) [inline pkg/fleet/installer/service/datadog_installer.go:193]:
```suggestion
	if os.IsNotExist(err) {
		return nil
	}
	return startUnit(ctx, installerUnit)
```

**Review by BaptisteFoy — APPROVED** [review]

File: pkg/fleet/installer/service/datadog_installer.go [hunk 3]
```diff
@@ -181,7 +181,16 @@ func getAgentIDs() (uid, gid int, err error) {
 
 // startInstallerStable starts the stable systemd units for the installer
 func startInstallerStable(ctx context.Context) (err error) {
-	return startUnit(ctx, installerUnit)
+	_, err = os.Stat("/etc/datadog-agent/datadog.yaml")
```
Comment (ogaca-dd) [inline pkg/fleet/installer/service/datadog_installer.go:184]: This doesnt' work on [MacOS](https://github.com/DataDog/datadog-agent/blob/03aedd2c7b82c9bd7b42512b8542e1e3a9cc347b/cmd/agent/common/path/path_darwin.go#L16) and on [FreeBSD](https://github.com/DataDog/datadog-agent/blob/03aedd2c7b82c9bd7b42512b8542e1e3a9cc347b/cmd/agent/common/path/path_freebsd.go#L16) but the build flag exclude only Windows (`//go:build !windows`).

Comment (arbll) [reply pkg/fleet/installer/service/datadog_installer.go:184]: Those platforms aren't supported by the installer today so it's fine 👍🏻

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**